### PR TITLE
8303137: [lworld] Fix ArchivedEnumTest fail caused by changed Enum static field

### PR DESCRIPTION
**Pull request by RogerRiggs**
Removed private static field in j.l.reflect.AccesFlag that CDS could not archive.
Replaced with inline expression.

---------
### Progress
- [x] Change must not contain extraneous whitespace

### Issue
 * [JDK-8303137](https://bugs.openjdk.org/browse/JDK-8303137): [lworld] Fix ArchivedEnumTest fail caused by changed Enum static field

### Reviewers
 * [Mandy Chung](https://openjdk.org/census#mchung) (@mlchung - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/valhalla pull/827/head:pull/827` \
`$ git checkout pull/827`

Update a local copy of the PR: \
`$ git checkout pull/827` \
`$ git pull https://git.openjdk.org/valhalla pull/827/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 827`

View PR using the GUI difftool: \
`$ git pr show -t 827`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/valhalla/pull/827.diff">https://git.openjdk.org/valhalla/pull/827.diff</a>

</details>

### CODE REVIEW

**Comment by mlchung** [comment]
Can you elaborate more what the issue is?  Why `SET_CLASS_IDENTITY_VALHALLA` is not an issue?

**Comment by RogerRiggs** [comment]
> Can you elaborate more what the issue is? Why `SET_CLASS_IDENTITY_VALHALLA` is not an issue?

The test is checking the 'current' value of the field to see if it is archived (or an exception).
The EMPTY_SET is an explicit exception in `cdsHeapVerifier.cpp`so I think it gets a pass.

I will remove both static fields and compute the values as the enums are created.

File: src/java.base/share/classes/java/lang/reflect/AccessFlag.java [hunk 1]
```diff
@@ -667,8 +665,6 @@ public enum Location {
         private static final Set<Location> SET_CLASS = Set.of(CLASS);
         private static final Set<Location> SET_CLASS_INNER_CLASS =
             Set.of(CLASS, INNER_CLASS);
-        private static final Set<Location> SET_CLASS_SUPER_VALHALLA =
-                ValhallaFeatures.isEnabled() ? EMPTY_SET : SET_CLASS_INNER_CLASS;
         private static final Set<Location> SET_CLASS_IDENTITY_VALHALLA =
                 ValhallaFeatures.isEnabled() ? SET_CLASS_INNER_CLASS : EMPTY_SET;
```
Comment (mlchung) [inline src/java.base/share/classes/java/lang/reflect/AccessFlag.java:669]: Should this static final field be removed?

Comment (RogerRiggs) [reply src/java.base/share/classes/java/lang/reflect/AccessFlag.java]: done

**Review by mlchung — APPROVED** [review]

**Pull request closed without merging**